requests_toolbelt/utils/dump.py

```diff
@@ -71,20 +169,24 @@ def _dump_request_data(request, prefixes, bytearr, proxy_info=None):
     bytearr.extend(prefix + b'Host: ' + host_header + b'\r\n')
 
     for name, value in headers.items():
+        value = sanitizer.request_header(name, value)
```

Comment:
Since our sanitizer will be sanitizing the entire headers dictionary, let's move this outside the loop